src/rules/selector-no-qualifying-type/__test__/index.js

```diff
@@ -0,0 +1,89 @@
+import testRule from "../../../testUtils/stylelint-test-rule-tape"
+import rule, { ruleName, messages } from ".."
+
+testRule(rule, {
+  ruleName,
+  config: [true],
+
+  accept: [ {
+    code: "input { top: 0; }",
+  }, {
+    code: ".class { top: 0; }",
+  }, {
+    code: "div>.class { top: 0; }",
+  }, {
+    code: "div+.class { top: 0; }",
+  }, {
+    code: "div~.class { top: 0; }",
+  }, {
+    code: "#id { top: 0; }",
+  }, {
+    code: "div>#id { top: 0; }",
+  }, {
+    code: "div+#id { top: 0; }",
+  }, {
+    code: "div~#id { top: 0; }",
+  } ],
+
+  reject: [ {
+    code: "input[type=\"button\"] { top: 0; }",
+    message: messages.rejected,
+  }, {
+    code: "div.class { top: 0; }",
+    message: messages.rejected,
+  }, {
+    code: "div#id { top: 0; }",
+    message: messages.rejected,
+  } ],
+})
```

Comment:
This scenario is currently disregarded. Think it is certainly required for the initial rule?